crates/holochain_p2p/tests/tests/node_messaging.rs

```diff
@@ -164,21 +164,74 @@ async fn test_call_remote() {
     let dna_hash = DnaHash::from_raw_36(vec![0; 36]);
     let handler = Arc::new(Handler::default());
 
-    let (agent1, _hc1) = spawn_test(dna_hash.clone(), handler.clone()).await;
-    let (_agent2, hc2) = spawn_test(dna_hash.clone(), handler).await;
+    let (agent1, hc1) = spawn_test(dna_hash.clone(), handler.clone()).await;
+    let (agent2, hc2) = spawn_test(dna_hash.clone(), handler).await;
+
+    tokio::time::timeout(std::time::Duration::from_secs(5), async {
+        loop {
+            tokio::time::sleep(std::time::Duration::from_millis(1)).await;
+
+            // make sure h2 has its own address
+            if hc2
+                .peer_store(dna_hash.clone())
+                .await
+                .unwrap()
+                .get_all()
+                .await
+                .unwrap()
+                .len()
+                > 0
+            {
+                break;
+            }
+        }
+
+        loop {
+            tokio::time::sleep(std::time::Duration::from_millis(1)).await;
```

Comment:
With that little sleep, there's a task for every remote call that's unanswered until hc1 becomes visible, right? Maybe increase the sleep?

Reply:
Perhaps... but the test runs in 0.026s, so I'm not concerned about it.

Reply:
Haha, right.